tools/local-environment/src/nomad.ts

```diff
@@ -672,216 +719,140 @@ export class Nomad {
-
-    const zippedDeploys = zip(oldCoreDeploys, oldBridgeDeploys);
+    const toEjectCores = [
+      ...updatedCores,
+      ...oldCoreDeploys.filter(
+        (oldCore) =>
+          !updatedCores.find(
+            (newCore) => newCore.chain.domain === oldCore.chain.domain
+          )
+      ),
+    ];
 
-    const delegated = true;
-    if (delegated) {
-      const promises = zippedDeploys.map(([oldCore, oldBridge]) => {
-        // We have a set of calls, but only here we know where they go. If we meld them into gnosis safe
-        const calls: CallData[] = [
-          getEnrollBridgeCall(newBridgeDeploy, oldBridge),
-          getEnrollReplicaCall(newCoreDeploy, oldCore),
-          ...getEnrollWatchersCall(newCoreDeploy, oldCore),
-        ];
+    const coreDeployArtifacts = this.ejectCoreDeploysArtifacts(toEjectCores);
 
-        if (oldCore == hostCoreDeploy) {
-          return governingRouter.callLocal(calls, hostCoreDeploy.overrides);
-        } else {
-          return governingRouter.callRemote(
-            oldCore.chain.domain,
-            calls,
-            hostCoreDeploy.overrides
-          );
-        }
-      });
+    const oldBridgeDeploys = this.getNetworks().map((n) => {
+      const core = this.getExistingCoreDeploy(n);
+      if (core) {
+        return this.getExistingBridgeDeploy(n)!;
+      }
+    });
 
-      await Promise.all([
-        ...promises,
-        // this also eventually should also be a prepared
-        // call which will be passed to itslef (GovernanceRouter) probably
-        governingRouter.setRouterGlobal(
-          newCoreDeploy.chain.domain,
-          toBytes32(newCoreDeploy.contracts.governance!.proxy.address)
+    const toEjectBridges = [
+      ...updatedBridges,
+      ...utils
+        .filterUndefined(oldBridgeDeploys)
+        .filter(
+          (oldBridge) =>
+            !updatedBridges.find(
+              (newBridge) => newBridge.chain.domain === oldBridge.chain.domain
+            )
         ),
-      ]);
-    } else {
-      // await Promise.all([
-      //     enrollCoreThroughGovernance(hostCoreDeploy, newCoreDeploy, oldCoreDeploys),
-      //     enrollBridgeThroughGovernance(hostCoreDeploy, newBridgeDeploy, oldBridgeDeploys),
-      //     enrollRouterThroughGovernance(hostCoreDeploy, newCoreDeploy),
-      // ])
-    }
+    ];
 
-    const remoteDomains = Array.from(this.deployed);
-    await checkCoreDeploy(newCoreDeploy, remoteDomains, host.domain);
-    await checkBridgeDeploy(newBridgeDeploy, remoteDomains);
-    await this.checkIncrementalDeploy(
-      newCoreDeploy,
-      newBridgeDeploy,
-      zippedDeploys
-    );
-
-    const coreDeployArtifacts = this.ejectCoreDeploysArtifacts([
-      newCoreDeploy,
-      ...oldCoreDeploys,
-    ]);
-    const bridgeDeployArtifacts = this.ejectBridgeDeploysArtifacts([
-      newBridgeDeploy,
-      ...oldBridgeDeploys,
-    ]);
+    const bridgeDeployArtifacts =
+      this.ejectBridgeDeploysArtifacts(toEjectBridges);
```

Comment:
i am confused what `eject` means here, should probably reword

Reply:
totally. Will reword in the next PR